core/server-role-manager/org.wso2.carbon.roles.mgt/src/main/java/org/wso2/carbon/roles/mgt/ServerRolesManager.java

```diff
@@ -137,21 +133,36 @@ public boolean addServerRoles(String[] serverRolesArray, String serverRoleType)
                 //todo manage duplicates - done(used Sets)
                 serverRolesList = ServerRoleUtils.mergeLists(serverRolesList, serverRolesListToAdd);
                 resource.setProperty(serverRoleType, serverRolesList);
-                status = true;
             }
             putResourceToRegistry(configReg, resource, regPath);
 
             // We have to update the modified flag in any case.. serverRoleType above
             // is always custom.
             String defaultPath = getRegistryPath(ServerRoleConstants.DEFAULT_ROLES_ID);
             Resource defaultResource = getResourceFromRegistry(configReg, defaultPath);
-            if (defaultResource != null) {
-                defaultResource.setProperty(ServerRoleConstants.MODIFIED_TAG,
-                        ServerRoleConstants.MODIFIED_TAG_TRUE);
+            if (defaultResource == null) {
+                List<String> productServerRolesList = ServerRoleUtils.readProductServerRoles();
+                try {
+                    defaultResource = this.createDefaultProductServerRoles(configReg, productServerRolesList);
+                } catch (RegistryException e) {
+                    this.handleException(e.getMessage(), e);
+                }
             }
+            defaultResource.setProperty(ServerRoleConstants.MODIFIED_TAG,
+                    ServerRoleConstants.MODIFIED_TAG_TRUE);
             putResourceToRegistry(configReg, defaultResource, defaultPath);
         }
-        return status;
+        return true;
```

Comment:
why are you always return true here. it should be true when server role added to the system correctly. but if serverRolesArray is null, we are not adding the server role.